libsolidity/ast/Types.h

```diff
@@ -736,6 +736,13 @@ class ArrayType: public ReferenceType
 
 	TypePointer copyForLocation(DataLocation _location, bool _isPointer) const override;
 
+	/// The offset to advance in calldata to move from one array element to the next.
```

Comment:
Please add caveat that this can only be used for statically encoded base types.

Reply:
It does exactly what I'd expect for dynamically encoded base types - it's the offset to advance from one element to the next, just that the elements in case of a dynamically encoded base type are themselves offsets/pointers.